controllers/webhook/webhook_controller.go

```diff
@@ -0,0 +1,220 @@
+			},
+		},
+	}
+
+	url, err := getReadinessProbeURL(clientConfig)
+	if err != nil {
+		return false, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return false, err
+	}
+
+	log.Info("Executing readiness probe on remote control plane", "url", req.URL.String())
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		log.Info("Probe failed", "error", err)
+		return false, err
+	}
+	log.Info("Probe response", "response", resp.StatusCode)
+
+	return resp.StatusCode == http.StatusOK, nil
+}
+
+func getReadinessProbeURL(config admissionv1.WebhookClientConfig) (string, error) {
+	switch {
+	case config.URL != nil:
+		return "", errors.New("only webhooks pointing to a Service are supported")
+
+	case config.Service != nil:
+		svc := config.Service
+		port := 443
+		if svc.Port != nil {
+			port = int(*svc.Port)
+		}
+		return fmt.Sprintf("https://%s.%s.svc:%d/ready", svc.Name, svc.Namespace, port), nil
+
+	default:
+		return "", errors.New("no URL or Service specified in WebhookClientConfig")
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		WithOptions(controller.Options{
+			LogConstructor: func(req *reconcile.Request) logr.Logger {
+				log := mgr.GetLogger().WithName("ctrlr").WithName("webhook")
+				if req != nil {
+					log = log.WithValues("MutatingWebhookConfiguration", req.Name)
+				}
+				return log
+			},
+		}).
+		For(&admissionv1.MutatingWebhookConfiguration{}, builder.WithPredicates(ownedByRemoteIstioPredicate(mgr.GetClient()))).
+		Complete(reconciler.NewStandardReconciler(r.Client, &admissionv1.MutatingWebhookConfiguration{}, r.Reconcile))
+}
+
+func ownedByRemoteIstioPredicate(cl client.Client) predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isOwnedByRemoteIstio(cl, e.Object)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isOwnedByRemoteIstio(cl, e.ObjectNew)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isOwnedByRemoteIstio(cl, e.Object)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return isOwnedByRemoteIstio(cl, e.Object)
+		},
+	}
+}
+
+func isOwnedByRemoteIstio(cl client.Client, obj client.Object) bool {
+	for _, ownerRef := range obj.GetOwnerReferences() {
+		if ownerRef.APIVersion == v1alpha1.GroupVersion.String() && ownerRef.Kind == v1alpha1.IstioRevisionKind {
+			rev := &v1alpha1.IstioRevision{}
+			err := cl.Get(context.Background(), client.ObjectKey{Name: ownerRef.Name}, rev)
+			if err != nil {
+				// TODO log error
+			} else if rev.Spec.Type == v1alpha1.IstioRevisionTypeRemote {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func getPeriod(webhook *admissionv1.MutatingWebhookConfiguration) time.Duration {
+	if period, ok := webhook.Annotations[constants.WebhookReadinessProbePeriodSecondsAnnotationKey]; ok {
```

Comment:
You can configure the probe by setting this annotation on the webhook? Couldn't these fields could be part of the RemoteIstio API instead?

Reply:
No, as explained above, the webhook's readiness status is managed by a separate controller, not the remoteistio controller.

Reply:
> This is because the status must be checked periodically and not when some owned resource is updated

This makes sense why it would be a separate controller but it would be nice to configure the webhook settings through the `RemoteIstio` CR rather than having to set the field on the webhook annotation if you wanted to change the probe interval. Is it possible to add a field to the `RemoteIstio` CR that will set this annotation on the webhook? e.g.

```
RemoteIstio:
  spec:
    webhook:
      probeInterval: 45s
```
-->
```
MutatingWebhook:
  metadata:
    annotations:
      operator.istio.io/readiness-probe-interval: 45s
```

I think the webhook controller would remain the same it's just that the knob to change the setting would be exposed on the `RemoteIstio`.

Reply:
That's a good idea, but I don't think we should implement it as proposed. Currently, we don't really have any configuration outside of `.spec.values` (apart from the version and upgrade strategy stuff), so we would have to add the readiness probe config to the values. But this would require upstream changes, which aren't needed and would also have no effect when deploying Istio via istioctl or Helm directly. 

So, I think the correct solution is to allow users to set these annotations as annotations instead of introducing a higher-level config that is then translated into annotations. But, currently, there is no `values.sidecarInjectorWebhook.annotations` field and I'm not sure if we should add them. We've gone back and forth whether standard k8s fields should be configurable via values or whether they should be set in helm post-processing (e.g. via kustomize). So, I suggest we defer this change until we decide how to handle these standard k8s fields.